src/main/kotlin/com/exactpro/th2/common/schema/message/ManualAckDeliveryCallback.kt

```diff
@@ -26,11 +26,22 @@ fun interface ManualAckDeliveryCallback {
      * @param delivery the delivered message
      * @param confirmProcessed the action that should be invoked when the message can be considered as processed
      */
+    @Deprecated(
+        "This method does not provide all necessary information about a message",
+        ReplaceWith("handle(deliveryMetadata, delivery, confirmProcessed)")
+    )
     @Throws(IOException::class)
     fun handle(consumerTag: String, delivery: Delivery, confirmProcessed: Confirmation)
 
-    fun interface Confirmation {
+    @Throws(IOException::class)
+    fun handle(deliveryMetadata: DeliveryMetadata, delivery: Delivery, confirmProcessed: Confirmation) {
```

Comment:
We should move this change to the ConfirmationMessageListener interface because it is part of public common API.
the handle method of this class should be extended via DeliveryMetadata